pkg/istio/processor/controller.go

```diff
@@ -80,16 +83,23 @@ func (c *Controller) processNextItem() bool {
 	err := c.sync(item)
 	if err != nil {
 		log.Errorf("%s processNextItem() Error performing %s for resource: %s: %s", logPrefix, item.Operation, item.Resource.Key(), err)
-		if item.ErrorHandler != nil {
-			err := item.ErrorHandler(err, item)
-			if err != nil && c.queue.NumRequeues(itemRaw) < queueNumRetries {
-				log.Infof("%s processNextItem() Retrying %s for resource: %s due to sync error", logPrefix, item.Operation, item.Resource.Key())
-				c.queue.AddRateLimited(itemRaw)
-				return true
-			}
-		}
+	}
+	if item.CallbackHandler == nil {
+		c.queue.Forget(itemRaw)
+		return true
 	}
 
+	// All errors/successes should be handled by the CallbackHandler()
+	err = item.CallbackHandler(err, item)
```

Comment:
Currently both of the controllers never return an error here. As we discussed we should add explicit error checks and retry here on network issues and potentially some api server errors.